tests/heuristic_calc_test.pl

```diff
@@ -0,0 +1,57 @@
+:- consult('../alphabeta/heuristic_calc.pl'). % import possible states impl.
+
+run:-
+    % test numberOfPawns
+
+    numberOfPawns(white, [piece(3,3,black,false),piece(4,4,black,false)], 0),
+    numberOfPawns(white, [piece(3,3,white,false),piece(4,4,black,false)], 1),
+    numberOfPawns(white, [piece(3,3,white,false),piece(4,4,white,true)], 1),
+    numberOfPawns(white, [piece(3,3,white,true),piece(4,4,white,true)], 0),
+    writeln('numberOfPawns - done!'),
+
+    numberOfQueens(white, [piece(3,3,black,false),piece(4,4,black,true)], 0),
+    numberOfQueens(white, [piece(3,3,white,false),piece(4,4,white,false)], 0),
+    numberOfQueens(white, [piece(3,3,white,false),piece(4,4,white,true)], 1),
+    numberOfQueens(white, [piece(3,3,white,true),piece(4,4,white,true)], 2),
+    writeln('numberOfQueens - done!'),
+
+    numberOfSafePawns(white, [piece(0,0,black,false),piece(5,4,black,true)], 6, 0),
```

Comment:
you dont have a test for unsafe queens

Reply:
meaning a white queen being threaten by a black piece (or a queen) and it the white turn

Reply:
"safe" - adjacent to the edge of the board and not threaten by another piece